### ISSUE TITLE
Not working on Play 2.5.0

### ISSUE THREAD

**Issue by 0xbaadf00d** [mohiva/play-html-compressor#49]
Hi,

It seem not working on Play 2.5 (with a project freshly created from activator 1.3.7)

```
"com.mohiva" %% "play-html-compressor" % "0.6.0",
``` 

```bash
[error] P:\Users\thiba\Documents\Programmation\identity-25\app\Filters.java:45:  com.mohiva.play.htmlcompressor.HTMLCompressorFilter cannot be converted to play.mvc.EssentialFilter
[error]         this.registeredFilters = new EssentialFilter[] {extraHeadersFilter, htmlCompressorFilter};
```

```java
@Singleton
public class Filters implements HttpFilters {

    /**
     * Current running environment.
     *
     * @since 16.03.23
     */
    private final Environment env;

    /**
     * Registered filters.
     *
     * @since 16.03.23
     */
    private final EssentialFilter[] registeredFilters;

    /**
     * Build an instance with injected variables.
     *
     * @param extraHeadersFilter   The extra headers filter instance
     * @param htmlCompressorFilter The HTML compressor filter instance
     * @since 16.03.10
     */
    @Inject
    public Filters(final Environment env, final HttpExtraHeadersFilter extraHeadersFilter, final HTMLCompressorFilter htmlCompressorFilter) {
        this.env = env;
        this.registeredFilters = new EssentialFilter[] {extraHeadersFilter, htmlCompressorFilter};
    }

    @Override
    public EssentialFilter[] filters() {
        return this.registeredFilters;
    }
}

```

**Comment by akkie** [comment]
Since Play 2.5 the Java API has a native filter API. Prior to Play 2.5 there was only a Scala API for filters. Now it seems that the Java and the Scala API's are not compatible.

Please see:
https://groups.google.com/forum/#!topic/play-framework-dev/CrSiDgDIhcY

**Comment by 0xbaadf00d** [comment]
It seem that using `.asJava()`:



```
@Inject
public Filters(final Environment env, final HttpExtraHeadersFilter extraHeadersFilter, final HTMLCompressorFilter htmlCompressorFilter) {
    this.env = env;
    this.registeredFilters = new EssentialFilter[] {extraHeadersFilter, htmlCompressorFilter.asJava()};
}
```

**Comment by akkie** [comment]
Thanks for the info. I've updated my mailing list question.

**Comment by mkurz** [comment]
@akkie A bit offtopic but I plan to upgrade to Play 2.5 soon and maybe use `play-html-compressor`:
* `play-html-compressor` is using Google's HTML Compressor  - where is it hosted? Google Code is dead AFAIK and https://code.google.com/archive/p/htmlcompressor/ looks like archived...
* What are the performance impacts of using `play-html-compressor`? I guess each response will run through the filter and gets compressed - will this cause any significant (measurable) performance impacts?

**Comment by akkie** [comment]
@mkurz I think the project itself is also dead. The latest release is from Oct, 2011. But it works. I've not measured the performance. So I cannot make a statement here.

**Comment by mkurz** [comment]
@akkie Thanks!

**Comment by akkie** [comment]
Fixed with #51
